Lib/csharp/csharphead.swg

```diff
@@ -126,11 +137,135 @@ SWIGEXPORT void SWIGSTDCALL SWIGRegisterExceptionArgumentCallbacks_$module(
+    unsafe static void SetPendingInvalidOperationException(char* message) {
+      SWIGPendingException.Set(new global::System.InvalidOperationException(new string(message), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionDelegate))]
+    unsafe static void SetPendingIOException(char* message) {
+      SWIGPendingException.Set(new global::System.IO.IOException(new string(message), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionDelegate))]
+    unsafe static void SetPendingNullReferenceException(char* message) {
+      SWIGPendingException.Set(new global::System.NullReferenceException(new string(message), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionDelegate))]
+    unsafe static void SetPendingOutOfMemoryException(char* message) {
+      SWIGPendingException.Set(new global::System.OutOfMemoryException(new string(message), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionDelegate))]
+    unsafe static void SetPendingOverflowException(char* message) {
+      SWIGPendingException.Set(new global::System.OverflowException(new string(message), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionDelegate))]
+    unsafe static void SetPendingSystemException(char* message) {
+      SWIGPendingException.Set(new global::System.SystemException(new string(message), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionArgumentDelegate))]
+    unsafe static void SetPendingArgumentException(char* message, char* paramName) {
+      SWIGPendingException.Set(new global::System.ArgumentException(new string(message), new string(paramName), SWIGPendingException.Retrieve()));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionArgumentDelegate))]
+    unsafe static void SetPendingArgumentNullException(char* message, char* paramName) {
+      global::System.Exception e = SWIGPendingException.Retrieve();
+      string decodedMessage = new string(message);
+      string decodedParamName = new string(paramName);
+      if (e != null) decodedMessage = decodedMessage + " Inner Exception: " + e.Message;
+      SWIGPendingException.Set(new global::System.ArgumentNullException(decodedParamName, decodedMessage));
+    }
+
+    [$imclassname.MonoPInvokeCallback(typeof(ExceptionArgumentDelegate))]
+    unsafe static void SetPendingArgumentOutOfRangeException(char* message, char* paramName) {
+      global::System.Exception e = SWIGPendingException.Retrieve();
+      string decodedMessage = new string(message);
+      string decodedParamName = new string(paramName);
+      if (e != null) decodedMessage = decodedMessage + " Inner Exception: " + e.Message;
+      SWIGPendingException.Set(new global::System.ArgumentOutOfRangeException(decodedParamName, decodedMessage));
+    }
```

Comment:
make a define `SWIG_STRING_BLIT_SAFENESS`

make it empty without AOT, make it `unsafe` with Mono-AOT